src/storage-types/src/connections.rs

```diff
@@ -358,6 +368,166 @@ impl<C: ConnectionAccess> AlterCompatible for Connection<C> {
     }
 }
 
+#[derive(Arbitrary, Clone, Debug, Eq, PartialEq, Hash, Serialize, Deserialize)]
+pub enum IcebergCatalogType {
+    Rest,
+    S3TablesRest,
+}
+
+#[derive(Arbitrary, Clone, Debug, Eq, PartialEq, Hash, Serialize, Deserialize)]
+pub struct IcebergCatalogConnection<C: ConnectionAccess = InlinedConnection> {
+    /// What type of Iceberg catalog is being used
+    pub catalog_type: IcebergCatalogType,
+    /// Where the catalog is located
+    pub uri: String,
+    /// Variably referred to as a warehouse or "catalog"
+    pub warehouse: Option<String>,
+    /// For REST catalogs, the oauth2 credential in a `CLIENT_ID:CLIENT_SECRET` format
+    pub credential: Option<StringOrSecret>,
+    /// The AWS connection details, for s3tables
+    pub aws_connection: Option<AwsConnectionReference<C>>,
+    /// The oauth2 scope for REST catalogs
+    pub scope: Option<String>,
```

Comment:
This really looks like an `enum` type. Can we move the fields that are specific to each type under the `IcebergCatalogType` enum so that there aren't optional?